oak_remote_attestation_verification/src/verifier.rs

```diff
@@ -97,39 +242,76 @@ pub fn verify_endorsement_statement(
     endorsement_bytes: &[u8],
     binary_digest: &[u8],
 ) -> anyhow::Result<()> {
-    let binary_digest = core::str::from_utf8(binary_digest)?;
     let claim = parse_endorsement_statement(endorsement_bytes)?;
     if let Err(err) = validate_endorsement(&claim) {
         anyhow::bail!("validating endorsement: {err:?}");
     }
+    verify_validity_duration(&claim)?;
     if claim.subject.len() != 1 {
         anyhow::bail!(
             "expected 1 subject in the endorsement, found {}",
             claim.subject.len()
         );
     }
-    if claim.subject[0].digest["sha256"] != binary_digest {
-        anyhow::bail!(
-            "unexpected binary SHA256 digest: expected {binary_digest}, got {}",
-            claim.subject[0].digest["sha256"]
-        );
+
+    // TODO(#3641): Instead of the if-block, fail if the binary digest is empty.
+    if !binary_digest.is_empty() {
+        let binary_digest = core::str::from_utf8(binary_digest)?;
+        if claim.subject[0].digest["sha256"] != binary_digest {
+            anyhow::bail!(
+                "unexpected binary SHA256 digest: expected {binary_digest}, got {}",
+                claim.subject[0].digest["sha256"]
+            );
+        }
     }
     Ok(())
 }
 
+pub struct RekorLogEntryVerifier {
+    base64_pem_encoded_rekor_public_key: String,
+    base64_pem_encoded_endorser_public_key: String,
+}
+
+impl AttestationVerifier for RekorLogEntryVerifier {
+    fn verify(
+        &self,
+        _evidence: &AttestationEvidence,
```

Comment:
Does it mean that the caller still needs to provide this argument, even if it's not used?

Reply:
Yes. We could make all arguments optional in the definition of `AttestationVerifier`, but I think in reality, when we create an instance of a Verifier starting from a  `ConfigurableAttestationVerifier` all arguments are present, even thought not all of them will be needed in specific verifiers.